TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ImplMecanumDrive.java

```diff
@@ -0,0 +1,144 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+import com.qualcomm.robotcore.util.ElapsedTime;
+
+/**
+ * algorithm for calculation of magnitude
+ * for use with advanced mecanum drives
+ * jlemke 9/26/2017 @ 7:26 AM
+ */
+
+@TeleOp(name="Mecanum Drive Test", group="TeleOp")
+public class ImplMecanumDrive extends OpMode
+{
+    // move that gear up
+    Hardware750 robot = new Hardware750();
+    // setup runtime timer
+    private ElapsedTime runtime = new ElapsedTime();
+
+    @Override
+    public void init() {
+        telemetry.addData("Status", "Uninitialized...");
+        robot.init(hardwareMap);
+        telemetry.addData("Status", "Initialized");
+    }
+
+    /*
+     * Code to run REPEATEDLY after the driver hits INIT, but before they hit PLAY
+     */
+    @Override
+    public void init_loop() {
+    }
+
+    @Override
+    public void start() {
+        runtime.reset();
+    }
+
+    @Override
+    public void loop() {
+        telemetry.addData("LeftTrig", gamepad1.left_trigger);
+        telemetry.addData("RightTrig", gamepad1.right_trigger);
+        telemetry.addData("lstick X", gamepad1.left_stick_x);
+        telemetry.addData("lstick Y", (-1 * gamepad1.left_stick_y));
+        double presquare = Math.pow(gamepad1.left_stick_x, 2) + Math.pow(gamepad1.left_stick_y, 2);
+        telemetry.addData("presquare", presquare);
+        double magnitude = ((-1 * gamepad1.left_stick_y) < 0) ? -1*Math.sqrt(presquare) : Math.sqrt(presquare);
+        if (magnitude > 1) {
+            magnitude = 1;
+        }
+        telemetry.addData("magnitude", magnitude);
+        double angle = Math.atan2((-1*gamepad1.left_stick_y),gamepad1.left_stick_x);
+        if (angle < 0) {
+            angle += Math.PI*2;
+        }
+        double triggerVal = (-1 * gamepad1.left_trigger) + gamepad1.right_trigger;
+        telemetry.addData("ang le", angle);
+        telemetry.addData("number of pi", (angle / Math.PI));
+        double extraPi = angle / Math.PI;
+        //Makes variables for the Voltage Magnitude
+        double VM1 = ((magnitude) * (Math.sin(extraPi + (Math.PI / 4))) + triggerVal);
+        double VM2 = ((magnitude) * (Math.cos(extraPi + (Math.PI / 4))) - triggerVal);
+        double VM3 = ((magnitude) * (Math.cos(extraPi + (Math.PI / 4))) + triggerVal);
+        double VM4 = (((magnitude) * (Math.sin(extraPi + (Math.PI / 4))) - triggerVal));
+        //finds the largest Voltage Magnitude above 1 or below -1 and records it
+        double [] thing = {1, VM1, VM2, VM3, VM4};
+        int idk = 0;
+        for (int i = 1; i < 5; i++) {
+          if ((Math.abs(thing [i]) > 1) && (Math.abs(thing [i]) > Math.abs(thing [i - 1]))) {
+                idk = i;
+          }
+        }
+        //Create an array
+        double [] VM20 = new double [4];
```

Comment:
Why do we suddenly create a new array? Also, this array will be created every time you loop. Be aware of this. I'd strongly recommend declaring variables outside of the loop.